app/controllers/api/base_controller/authentication.rb

```diff
@@ -127,6 +136,14 @@ def validate_system_token_timestamp(timestamp)
         raise "Missing timestamp" if timestamp.blank?
         raise "Invalid timestamp #{timestamp} specified" if SYSTEM_TOKEN_TTL.ago.utc > timestamp
       end
+
+      def valid_ui_session?
+        [
+          valid_authenticity_token?(session, request.x_csrf_token), # CSRF token be set and valid
+          session[:userid].present?,                                # session has a userid stored
```

Comment:
Does the api worker have access to the session?

Reply:
@Fryguy yes, first I wanted to reinvent the wheel with connecting to memcached and all the sugar around, but it actually works. Tested on a live appliance...